package editor.protocol;

import "editor_settings.proto";
import "common.proto";

message Style
{
	optional string id = 1;

	optional common.protocol.Color4 fill_color = 2;
	optional common.protocol.Color4 outline_color = 3;
	optional float outline_thickness = 4 [default = 0];

    // Regular    = 0,
    // Bold       = 1 << 0,
    // Italic     = 1 << 1,
    // Underlined = 1 << 2
    optional uint32 font_style = 5;
}

message Styles
{
	repeated Style styles = 1;
}